src/Misc/layoutbin/powershell/Add-AzureGuestAgentCapabilities.ps1

```diff
@@ -0,0 +1,24 @@
+[CmdletBinding()]
+param()
+
+$guestAgentPath = $null
+
+try {
+    $proc = Get-Process 'WindowsAzureGuestAgent'
```

Comment:
Depending on the current security context, processes from other users may not be visible. It sounds like you have thought through that already and are comfortable with that implication for your scenario.

Also it may be good to be explicit and use named parameter syntax (`Get-Process -Name 'WindowsAzureGuestAgent'`) rather than rely on positional parameter. Although given what cmdlet this is (Get-Process), it's hard to imagine the simple scenario `Get-Process '<NAME>'` would break in the future. So positional is probably fine too.

fyi only:
One thing you can do to simplify this is `Get-Process -Name 'WindowsAzureGuestAgent' | Select-Object -First 1 | ForEach-Object { Write-Capability -Name 'AzureGuestAgent' -Value $_.Path }` rather than capturing value and branching depending on whether array or not. Also PowerShell has a language feature for dealing with this type of problem: `$result = @(Get-Process -Name 'WindowsAzureGuestAgent')` will always return an array, so then you can simply `if ($result) { ... $result[0] }`

Also another thing to simplify would be to use `-ErrorAction Ignore` instead of try-catch.